haskell-src/Concordium/ID/Parameters.hs

```diff
@@ -27,6 +30,30 @@ foreign import ccall safe "global_context_to_bytes" globalContextToBytes :: Ptr
 foreign import ccall safe "global_context_from_bytes" globalContextFromBytes :: Ptr Word8 -> CSize -> IO (Ptr GlobalContext)
 foreign import ccall safe "global_context_to_json" globalContextToJSONFFI :: Ptr GlobalContext -> Ptr CSize -> IO (Ptr Word8)
 foreign import ccall safe "global_context_from_json" globalContextFromJSONFFI :: Ptr Word8 -> CSize -> IO (Ptr GlobalContext)
+foreign import ccall unsafe "global_context_create"
+    createGlobalContextFFI ::
+        Ptr Word8 -> CSize -> -- Pointer to a byte array representing the genesis string and its length.
+                              -- The array must correspond to a utf8 encoded string.
+        Ptr Word8 -> CSize -> -- Pointer to a byte array and its length. The array must be the binary
+                              -- representation of a `Generators<G1>` Rust instance.
+        Ptr Word8 -> CSize -> -- Pointer to a byte array and its length. The array must be the binary
+                              -- representation of a `PedersenKey<G1>` Rust instance.
+        IO (Ptr GlobalContext)
+
+-- Create an GlobalContext intance from bytestrings and texts.
+-- This function is a wrapper for `createGlobalContextFFI`, and is used for creating heap-allocated
+-- `GlobalContext` instances from bytestrings. See the `createGlobalContextFFI` import declaration
+-- for information about the function inputs and preconditions.
+createGlobalContext :: Text -> BS8.ByteString -> BS8.ByteString -> GlobalContext
+createGlobalContext genString bulletProofGens onChainComm = unsafePerformIO ( do
+    ptr <- unsafeUseAsCStringLen (Text.encodeUtf8 genString) $ \(gsPtr, gsLen) ->
+        unsafeUseAsCStringLen bulletProofGens $ \(bpgPtr, bpgLen) ->
+            unsafeUseAsCStringLen onChainComm $ \(occPtr, occLen) ->
```

Comment:
It is important to be aware that if the string is empty what the Rust side gets is an arbitrary pointer. So it is important to document that this situation si handled appropriately on the Rust side. `slice_from_c_bytes` handles it correctly. But please state this as we do in other cases.